src/Workload/Microsoft.Maui.Sdk/Sdk/BundledVersions.in.targets

```diff
@@ -14,54 +15,23 @@
-    <_MauiImplicitPackageReference Include="Microsoft.Maui.Resizetizer" Version="$(MauiVersion)" PrivateAssets="all" />
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMauiEssentials)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.Maui.Essentials" Version="$(MauiVersion)">
-      <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
-    </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMauiCore)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.Maui.Core" Version="$(MauiVersion)">
-      <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
-    </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMaui)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)">
-      <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
-    </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMaui)' == 'true' and '$(UseMauiCompat)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)">
-      <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
-    </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <ItemGroup Condition=" '$(UseMaui)' == 'true' and '$(UsingMicrosoftNETSdkRazor)' == 'true' ">
-    <_MauiImplicitPackageReference Include="Microsoft.AspNetCore.Components.WebView.Maui" Version="$(MauiVersion)">
-      <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
-    </_MauiImplicitPackageReference>
-  </ItemGroup>
-  <!-- Always install the required dependecies, such as build tools for the SDK itself -->
-  <ItemGroup>
-    <_MauiRequiredImplicitPackageReference Remove="@(PackageReference)" />
-    <PackageReference Include="@(_MauiRequiredImplicitPackageReference)" />
-  </ItemGroup>
-  <!-- Install the nuget packages if we are supposed to -->
-  <ItemGroup Condition=" '$(UseMauiNuGets)' != 'true' ">
-    <_MauiImplicitPackageReference Remove="@(PackageReference)" />
-    <PackageReference Include="@(_MauiImplicitPackageReference)" />
+  <ItemGroup Condition=" '$(EnableMauiImplicitPackageReferences)' != 'false' and '$(UseMauiNuGets)' != 'true' ">
+    <_MauiAddedImplicitPackageReference Include="@(PackageReference->HasMetadata('IsMauiImplicitPackageReference'))" />
+    <_MauiAddedUserPackageReference Include="@(PackageReference->WithMetadataValue('IsMauiImplicitPackageReference',''))" />
+    <_MauiMissingImplicitPackageReference Include="@(_MauiAddedImplicitPackageReference)" Exclude="@(_MauiAddedUserPackageReference)" />
+    <PackageReference Remove="@(PackageReference)" />
```

Comment:
Is there any chance this is too broad of an item group? Should we filter this first to any Microsoft.Maui or other known patterns (androidx, etc) that we for sure know were added by us implicitly?

Reply:
I am using the `IsMauiImplicitPackageReference` metadata that I have attached to the items

Reply:
OK, so I updated the logic to only touch packages that we have added/marked as maui-specific. As a post fix step, we remove all Maui packages, then add back the user ones - and then add back the missing implicit ones. This reduces the issues with non-maui packages.